include/exadg/acoustic_conservation_equations/spatial_discretization/operators/weak_boundary_conditions.h

```diff
@@ -0,0 +1,165 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef EXADG_ACOUSTIC_CONSERVATION_EQUATIONS_SPATIAL_DISCRETIZATION_OPERATORS_WEAK_BOUNDARY_CONDITIONS_H_
+#define EXADG_ACOUSTIC_CONSERVATION_EQUATIONS_SPATIAL_DISCRETIZATION_OPERATORS_WEAK_BOUNDARY_CONDITIONS_H_
+
+#include <exadg/acoustic_conservation_equations/user_interface/boundary_descriptor.h>
+#include <exadg/functions_and_boundary_conditions/evaluate_functions.h>
+#include <exadg/matrix_free/integrators.h>
+
+namespace ExaDG
+{
+namespace Acoustics
+{
+/**
+ * Class to access boundary values with a similar interface to FaceIntegrators.
+ * This way, the operations on faces only has to be written once.
+ */
+template<int dim, typename Number>
+class BoundaryFaceIntegratorP
+{
+  using FaceIntegratorP = FaceIntegrator<dim, 1, Number>;
+
+public:
+  BoundaryFaceIntegratorP(FaceIntegratorP const &          integrator_m_in,
+                          BoundaryDescriptorP<dim> const & boundary_descriptor_in)
+    : integrator_m(integrator_m_in),
+      boundary_descriptor(boundary_descriptor_in),
+      evaluation_time(0.0),
+      boundary_id(dealii::numbers::invalid_boundary_id),
+      boundary_type(BoundaryTypeP::Undefined)
+  {
+  }
+
+  void
+  reinit(unsigned int const face)
+  {
+    auto const boundary_id_new = integrator_m.get_matrix_free().get_boundary_id(face);
+
+    // only update boundary_type if needed to avoid an unnecessary search in boundary_descriptor
+    if(boundary_id_new != boundary_id)
+    {
+      boundary_id   = boundary_id_new;
+      boundary_type = boundary_descriptor.get_boundary_type(boundary_id);
+    }
+  }
+
+  void
+  set_time(Number const time)
+  {
+    evaluation_time = time;
+  }
+
+  inline DEAL_II_ALWAYS_INLINE //
+    typename FaceIntegratorP::value_type
+    get_value(unsigned int const q) const
+  {
+    if(boundary_type == BoundaryTypeP::Dirichlet)
+    {
+      auto const g = FunctionEvaluator<0, dim, Number>::value(
+        *boundary_descriptor.dirichlet_bc.find(boundary_id)->second,
+        integrator_m.quadrature_point(q),
+        evaluation_time);
+      return -integrator_m.get_value(q) + Number{2.0} * g;
+    }
+    else
+    {
+      AssertThrow(false,
+                  dealii::ExcMessage("Boundary type of face is invalid or not implemented."));
+    }
```

Comment:
Here, one could call a free function `get_exterior_value()` like for other operators in ExaDG.